src/app/redux/TransactionReducer.js

```diff
@@ -0,0 +1,187 @@
+import {fromJS, Map} from 'immutable';
+
+// Action constants
+const CONFIRM_OPERATION = 'transaction/CONFIRM_OPERATION';
+const HIDE_CONFIRM = 'transaction/HIDE_CONFIRM';
+export const BROADCAST_OPERATION = 'transaction/BROADCAST_OPERATION';
+export const UPDATE_AUTHORITIES = 'transaction/UPDATE_AUTHORITIES';
+export const UPDATE_META = 'transaction/UPDATE_META';
+const ERROR = 'transaction/ERROR';
+const DELETE_ERROR = 'transaction/DELETE_ERROR';
+const SET = 'transaction/SET';
+const REMOVE = 'transaction/REMOVE';
+// Saga-related
+export const RECOVER_ACCOUNT = 'transaction/RECOVER_ACCOUNT';
+
+const defaultState = fromJS({
+    operations: [],
+    status: { key: '', error: false, busy: false, },
+    errors: null
+});
+
+export default function reducer(state = defaultState, action) {
+    const payload = action.payload;
+
+    switch (action.type) {
+        case 'transaction/CONFIRM_OPERATION': {
+            const operation = fromJS(payload.operation);
+            const confirm = payload.confirm;
+            const warning = payload.warning;
+            return state.merge({
+                show_confirm_modal: true,
+                confirmBroadcastOperation: operation,
+                confirmErrorCallback: payload.errorCallback,
+                confirm,
+                warning
+            });
+        }
+
+        case 'transaction/HIDE_CONFIRM':
+            return state.merge({show_confirm_modal: false, confirmBroadcastOperation: undefined, confirm: undefined});
+
+        case 'transaction/BROADCAST_OPERATION':
+            // See TransactionSaga.js
+            return state;
+
+        case 'transaction/UPDATE_AUTHORITIES':
+            return state;
+
+        case 'transaction/UPDATE_META':
+            return state;
+
+        case 'transaction/ERROR': {
```

Comment:
The regexes in this block aren't going to change. They should be consts with nice descriptive names.

Reply:
mind opening an issue for this?

Reply:
actually it's covered already by https://github.com/steemit/condenser/issues/2037 possibly -- leave a comment pointing to this code

Reply:
Added, though I think it'd make sense to create the consts now. Let the testing be focused on tests.